src/panel/direction/DirectionPanel.jsx

```diff
@@ -125,7 +125,7 @@ export default class DirectionPanel extends React.Component {
           fire('set_route', { routes, vehicle, origin, destination });
         });
       } else {
-        this.setState({ isLoading: true, error: true });
+        this.setState({ isLoading: true, error: true, errorCode: directionResponse });
```

Comment:
I don't understand how the errorCode is in the `directionResponse` here :thinking: I thought it would be in the `errorCode` value of the object… Ensuring the shape of the return is always the same would prevent that.
It would also simplify the test of a valid response.